starsim/distributions.py

```
@@ -341,64 +353,99 @@ def make_dist(self, size, uids=None):
-                val = resized
-                kwds[key] = val # Replace 
+            if np.iterable(val) and uids is not None and (len(val) == len(uids)) and self.dist != 'choice': # TODO: figure out logic
+                self.has_array_pars = True
+            # else:
+            #     array
+            #     resized = np.zeros(size, dtype=val.dtype) # TODO: fix, problem from when there happen to be uid entries, but it's not slots
+            #     resized[uids] = val[:len(uids)] # TODO: check if slicing is ok here
+            #     val = resized
+            #     kwds[key] = val # Replace 
             
         return dist, kwds
     
-    def rvs(self, size=1, uids=None):
-        """ Main method for getting random variables """
+    def rvs(self, n=1, size=None): # TODO: fix!!!
+        """
+        Get random variables
+        
+        Args:
+            n (int/tuple/arr): if an int or tuple, return this many random variables; if an array, treat as UIDs
+        """
+        if size is not None:
+            n = size
+            size = None
         
         # Check for readiness
         if not self.initialized:
             raise DistNotInitializedError(self)
         if not self.ready and self.strict:
             raise DistNotReadyError(self)
         
-        # Shortcut if nothing to return
-        if not np.isscalar(size):
-            errormsg = f'Expecting a scalar size, not {size}; multidimensional output not supported and for UIDs, use urvs() instead'
-            raise ValueError(errormsg)
+        # Figure out size, UIDs, and slots
+        if np.isscalar(n) or isinstance(n, tuple):  # If passing a non-scalar size, interpret as dimension rather than UIDs iff a tuple
+            uids = None
+            slots = None
+            size = n
+        else:
+            uids = np.asarray(n)
+            if len(uids):
+                if self.slots is None:
+                    print('TEMP SLOTS')
```

Remember to remove before final.

---

Does the code ever hit this point in practice? The user provided UIDs, but we don't have slots. Might want to throw an exception rather than trying to continue with UIDs as slots?

---

It hit it in testing only, but I've refactored the tests so this is no longer needed and has been removed.